File: Programowanie_Obiektowe/PO_C/PO_C/Wilcza_Jagoda.cpp
#include "Wilcza_Jagoda.h"

Wilcza_Jagoda::Wilcza_Jagoda(int x, int y, Swiat& swiat) : Roslina(WILCZA_JAGODA_SILA, x, y, swiat) {}

void Wilcza_Jagoda::kolizja(Organizm& atakujacy)
{
    swiat.setLog(swiat.getLog() + string(1, rysowanie()) + " (" + to_string(x) + ", " + to_string(y) + ") zjedzony przez " + string(1, atakujacy.rysowanie()) + " (" + to_string(atakujacy.getX()) + ", " + to_string(atakujacy.getY()) + ")" + "\n");
    swiat.setLog(swiat.getLog() + string(1, atakujacy.rysowanie()) + " (" + to_string(atakujacy.getX()) + ", " + to_string(atakujacy.getY()) + ") zatruty przez " + string(1, rysowanie()) + "\n");

    swiat.ustawOrganizmNa(atakujacy.getY(), atakujacy.getX(), nullptr);
    swiat.ustawOrganizmNa(y, x, nullptr);

    swiat.usunOrganizm(&atakujacy);
    swiat.usunOrganizm(this);
}

char Wilcza_Jagoda::rysowanie() const
{
    return WILCZA_JAGODA;
}

Organizm* Wilcza_Jagoda::stworzPotomka(int x, int y) const
{
    return new Wilcza_Jagoda(x, y, swiat);
}

File: Programowanie_Obiektowe/PO_JAVA/src/Wilcza_Jagoda.java
public class Wilcza_Jagoda extends Roslina
{
    public Wilcza_Jagoda(int x, int y, Swiat swiat)
    {
        super(Stale.WILCZA_JAGODA_SILA, x, y, swiat);
    }

    @Override
    public void kolizja(Organizm atakujacy)
    {
        swiat.dodajLog(rysowanie() + " (" + x + ", " + y + ") zjedzony przez " + atakujacy.rysowanie() + " (" + atakujacy.getX() + ", " + atakujacy.getY() + ")");
        swiat.dodajLog(atakujacy.rysowanie() + " (" + atakujacy.getX() + ", " + atakujacy.getY() + ") zatruty przez " + rysowanie());

        swiat.ustawOrganizmNa(atakujacy.getX(), atakujacy.getY(), null);
        swiat.ustawOrganizmNa(x, y, null);

        swiat.usunOrganizm(atakujacy);
        swiat.usunOrganizm(this);
    }

    @Override
    public char rysowanie()
    {
        return Stale.WILCZA_JAGODA;
    }

    @Override
    public String nazwa()
    {
        return Stale.WILCZA_JAGODA_S;
    }

    @Override
    public Organizm stworzPotomka(int x, int y)
    {
        return new Wilcza_Jagoda(x, y, swiat);
    }
}